### ISSUE TITLE
qvm-template reinstall broken - qvm-template: error: Same version of template 'whonix-gw-16' not found.

### ISSUE THREAD

**Issue by adrelanos** [QubesOS/qubes-issues#7250]
### Qubes OS release

R4.1

### Brief summary

Template re-installation no longer possible in Qubes R4.1.

### Steps to reproduce

1. install Qubes 4.1, select install Whonix option
2. try Whonix template re-installation

### Actual behavior

> qvm-template list | grep whonix

```
whonix-gw-16  0:4.0.6-202201160402  @commandline
whonix-ws-16  0:4.0.6-202201160402  @commandline
```

`0:4.0.6-202201160402` really is latest version at time of writing as per https://github.com/QubesOS/qubes-issues/issues/6890#issuecomment-1031429983.

> qvm-template upgrade whonix-gw-16

```
Template 'whonix-gw-16' of highest version already installed, skipping...
```

> qvm-template reinstall whonix-gw-16

```
usage: qvm-template [--verbose] [--quiet] [--help] [--repo-files REPO_FILES] [--keyring KEYRING] [--updatevm UPDATEVM]
                    [--enablerepo REPOID] [--disablerepo REPOID] [--repoid REPOS] [--releasever RELEASEVER] [--refresh]
                    [--cachedir CACHEDIR] [--keep-cache] [--yes]
                    {install,reinstall,downgrade,upgrade,download,list,info,search,remove,purge,clean,repolist} ...
qvm-template: error: Same version of template 'whonix-gw-16' not found.
```

### Expected behavior

Template should be reinstalled after attempting both, `qvm-template upgrade` and `qvm-template reinstall`.

Previously this could be done in Qubes R4.0 using:

```
sudo qubes-dom0-update --enablerepo=qubes-templates-community --action=upgrade <qubes-template-package>
sudo qubes-dom0-update --enablerepo=qubes-templates-community --action=reinstall <qubes-template-package>
```

By running the above two commands, template would be re-installed.

### Whonix specific vs Whonix unspecific

While this example uses Whonix, I highly doubt it's a Whonix specific issue.

Therefore currently https://www.whonix.org/wiki/Qubes/Reinstall no longer works.

### Related

https://github.com/QubesOS/qubes-issues/issues/4518

**Comment by unman** [comment]
Confirmed that this is not Whonix specific - same behaviour with Debian templates

**Comment by marmarek** [comment]
> `sudo qubes-dom0-update --enablerepo=qubes-templates-community --action=upgrade <qubes-template-package>`
> By running the above two commands, template would be re-installed.

Should it really, when requesting an update and you have the latest version already?

**Comment by adrelanos** [comment]
Some way is very much desirable to re-install the template no matter if there is a newer one or not. Just for the sake of re-installing the template in case (user) error and better going back to a fresh template.

(If there is a newer template, install the newer one of course.)

Otherwise uninstall template + install template is a much more cumbersome method.
The current state is a R4.0 -> R4.1 usability regression.

This is related to https://github.com/QubesOS/qubes-issues/issues/4518 in summary "Never mind the currently installed version. I want to get rid of the current installed template. Replace it with the original from the repository." That wasn't possible but now there's no way to re-install an template at all without previously uninstalling it (which requires fiddling with VM settings to be able to uninstall it and then re-setup after new installation).

In other words, now `--reinstall` is clearly broken. Not sure what it could still be used for now.

> qvm-template: error: Same version of template 'whonix-gw-16' not found.

But there really is the really same version for that template in the repository.

**Comment by Narvey** [comment]
Is there a workaround for this? I tried deleting the template (**Don't do that!!!!**), and now it can't find it to re-install it. So that idea for a workaround is a dismal failure.

**Comment by adrelanos** [comment]
In Qubes R4.2, `upgrade` works for me.

**Comment by DemiMarie** [comment]
Does that mean this has been fixed @adrelanos?

**Comment by adrelanos** [comment]
In R4.1, I don't know.

In R4.2, fixed.

**Comment by ben-grande** [comment]
Tested and affects R4.3.
